.github/workflows/memory-benchmark.yml

```
@@ -334,6 +345,160 @@ jobs:
+          set -euo pipefail
+          cd services/memory
+          uv run python ../../benchmarks/memory/write_pipeline/bench_write_pipeline.py
+
+      - name: Enforce write-pipeline regression gate
+        run: |
+          cd services/memory
+          uv run python ../../benchmarks/memory/write_pipeline/regression_gate.py
+
+      - name: Upload write-pipeline artifact
+        if: always()
+        uses: actions/upload-artifact@043fb46d1a93c77aae656e7c1c64a875d1fc6a0a # v7.0.1
```

actionlint (1.7.12)

[error] unexpected key "uses" for step to run shell command. expected one of "continue-on-error", "env", "id", "if", "name", "run", "shell", "timeout-minutes", "working-directory"

(syntax-check)